### PR TITLE
Update pyproject-toml to support PEP 639

### LINKED ISSUE #13869: Support for PEP 639 in RUF200
In `--preview`, the following error is shown for PEP 639 style licenses:

```
tests/packages/spdx/pyproject.toml:5:17: RUF200 Failed to parse pyproject.toml: wanted string or table
  |
3 | version = "1.2.3"
4 | license = "MIT OR GPL-2.0-or-later OR (FSFUL AND BSD-2-Clause)"
5 | license-files = ["LICEN[CS]E*", "AUTHORS*", "licenses/LICENSE.MIT"]
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ RUF200
```

I don't fully understand the error message (`license-files` was never a string or table), but it would be good to support PEP 639, at least before this leaves preview.

### PR DESCRIPTION
**Pull request by konstin**
PEP 639 is provisionally accepted, we were previously shipping definitions from an earlier draft.

We don't validate all that's in the PEP (`license` vs. `license-files` compatibility, SPDX expressions), only the overall structure of pyproject.toml.

Updates pep440_rs in the process.

Fixes #13869

### CODE REVIEW

File: crates/ruff_linter/src/rules/ruff/snapshots/ruff_linter__rules__ruff__tests__RUF200_pep639.snap [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+source: crates/ruff_linter/src/rules/ruff/mod.rs
```
Comment (konstin) [inline crates/ruff_linter/src/rules/ruff/snapshots/ruff_linter__rules__ruff__tests__RUF200_pep639.snap:2]: This now became a valid example

File: crates/ruff_linter/src/rules/ruff/snapshots/ruff_linter__rules__ruff__tests__RUF200_various_invalid.snap [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+source: crates/ruff_linter/src/rules/ruff/mod.rs
+---
+pyproject.toml:10:16: RUF200 Failed to parse pyproject.toml: Expected package name starting with an alphanumeric character, found '.'
```
Comment (konstin) [inline crates/ruff_linter/src/rules/ruff/snapshots/ruff_linter__rules__ruff__tests__RUF200_various_invalid.snap:4]: Currently, we only emit a single error when various things are broken

File: crates/ruff_linter/src/rules/ruff/snapshots/ruff_linter__rules__ruff__tests__RUF200_invalid_author.snap [hunk 3]
```diff
@@ -1,13 +1,13 @@
 ---
 source: crates/ruff_linter/src/rules/ruff/mod.rs
 ---
-pyproject.toml:6:84: RUF200 Failed to parse pyproject.toml: invalid type: integer `1`, expected a string
+pyproject.toml:5:11: RUF200 Failed to parse pyproject.toml: a table with 'name' and/or 'email' keys
```
Comment (konstin) [inline crates/ruff_linter/src/rules/ruff/snapshots/ruff_linter__rules__ruff__tests__RUF200_invalid_author.snap:4]: The error message isn't ideal but the span should sufficiently prompt users to fix their authors table

**Comment by MichaReiser** [comment]
I went ahead and disabled the `pyproject_toml` linting for wasm to unblock this PR. That's why I request review from someone else.

**Comment by dhruvmanila** [comment]
Looking at the ecosystem changes, it seems that we'll now create diagnostic for the old definition related to `license-files`: https://github.com/apache/airflow/blob/476e77d9aba492a2159b256bc7a4aa86e560dee5/clients/python/pyproject.toml#L27. Do we want to maintain backwards compatibility?

**Comment by konstin** [comment]
I don't think so, that syntax was only part of a draft and never of any (provisionally) accepted PEP. It makes sense to lint against it to migrate the early adopters to the final version (and i'm expecting that hatch will migrate, too).

**Comment by MichaReiser** [comment]
Let's make this a breaking change.

**Pull request merged**